### PR TITLE
improve windows builds

### PR DESCRIPTION
**Pull request by shogo82148**

### CODE REVIEW

File: versions/win32.json [hunk 1]
```diff
@@ -40,17 +40,53 @@
...
+    "5.13.9",
+    "5.13.8",
+    "5.13.7",
+    "5.13.6",
+    "5.13.5",
+    "5.13.4",
+    "5.13.3",
+    "5.13.2",
+    "5.13.1",
+    "5.13.0",
```
Comment (shogo82148) [inline versions/win32.json:54]: TODO: delete me before merging

File: versions/win32.json [hunk 2]
```diff
@@ -40,17 +40,53 @@
...
     "5.12.3",
     "5.12.2",
     "5.12.1",
     "5.12.0",
+    "5.11.5",
+    "5.11.4",
+    "5.11.3",
+    "5.11.2",
+    "5.11.1",
+    "5.11.0",
```
Comment (shogo82148) [inline versions/win32.json:66]: TODO: delete me before merging

File: versions/win32.json [hunk 3]
```diff
@@ -40,17 +40,53 @@
...
     "5.10.1",
     "5.10.0",
+    "5.9.5",
+    "5.9.4",
+    "5.9.3",
+    "5.9.2",
+    "5.9.1",
+    "5.9.0",
```
Comment (shogo82148) [inline versions/win32.json:74]: TODO: delete me before merging

File: versions/win32.json [hunk 4]
```diff
@@ -40,17 +40,53 @@
...
+    "5.8.5",
+    "5.8.4",
+    "5.8.3",
+    "5.8.2",
+    "5.8.1",
+    "5.8.0",
+    "5.7.3",
+    "5.7.2",
+    "5.7.1",
+    "5.7.0",
```
Comment (shogo82148) [inline versions/win32.json:88]: TODO: delete me before merging

File: scripts/lib/Devel/PatchPerl/Plugin/MinGWGNUmakefile.pm [hunk 5]
```diff
@@ -0,0 +1,11840 @@
...
+MAKEFILE
+    if (_ge($version, "5.10.1")) {
+        _patch_gnumakefile($version, <<'PATCH');
+--- win32/GNUmakefile
++++ win32/GNUmakefile
+@@ -2,6 +2,7 @@
+ GCCBIN := gcc
+ INST_DRV := c:
+ INST_TOP := $(INST_DRV)\perl
++#WIN64		:= undef
```
Comment (shogo82148) [inline scripts/lib/Devel/PatchPerl/Plugin/MinGWGNUmakefile.pm:6973]: remove it

File: scripts/lib/Devel/PatchPerl/Plugin/MinGWGNUmakefile.pm [hunk 6]
```diff
@@ -0,0 +1,11840 @@
...
+ #INST_VER	:= \__INST_VER__
+ #INST_ARCH	:= \$(ARCHNAME)
+ #USE_SITECUST	:= define
+@@ -10,8 +11,6 @@
+ USE_IMP_SYS	:= define
+ USE_PERLIO	:= define
+ USE_LARGE_FILES	:= define
+-#USE_64_BIT_INT	:= define
+-#USE_LONG_DOUBLE :=define
```
Comment (shogo82148) [inline scripts/lib/Devel/PatchPerl/Plugin/MinGWGNUmakefile.pm:6982]: remove it

File: scripts/lib/Devel/PatchPerl/Plugin/MinGWGNUmakefile.pm [hunk 7]
```diff
@@ -0,0 +1,11840 @@
...
+ CCTYPE		:= GCC
+ #CFG		:= Debug
+ #USE_PERLCRT	= define
+@@ -20,6 +19,8 @@
+ #CRYPT_LIB     = -lfcrypt
+ #PERL_MALLOC	:= define
+ #DEBUG_MSTATS	:= define
++#BUILD_STATIC	:= define
++#ALL_STATIC	:= define
```
Comment (shogo82148) [inline scripts/lib/Devel/PatchPerl/Plugin/MinGWGNUmakefile.pm:6991]: remove it

File: .github/workflows/win32.yml [hunk 8]
```diff
@@ -98,7 +98,7 @@ jobs:
         env:
           PERL_VERSION: ${{ matrix.perl }}
           PERL5LIB: ${{ github.workspace }}/scripts/windows/local/lib/perl5
-          # PERL_DL_DEBUG: "10" # enables debugging
+          PERL_DL_DEBUG: "10" # enables debugging
```
Comment (shogo82148) [inline .github/workflows/win32.yml:101]: TODO: remove me

**Comment by tobyink** [comment]
It would be really helpful if there could be non-threaded builds for Windows. If I'm defining test cases in a matrix like:

```yaml
platform: [ 'ubuntu-latest', 'macos-latest', 'windows-latest' ]
perl:     [ '5.22', '5.24', '5.26', '5.28', '5.30', '5.32' ]
threaded: [ true, false ]
```

It's annoying to have to add exceptions for Windows threaded Perls.

**Comment by shogo82148** [comment]
not-threaded perl on Windows are very rare environment.
I can't guarantee to add it, but I'll try.

Before that, I have to make all builds green...

**Pull request merged**